lib/chromic_pdf/pdf/template.ex

```diff
@@ -0,0 +1,172 @@
+  };
+  </script>
+  """
+
+  @doc """
+  Returns source and options for a PDF to be printed, a given set of template options. The return
+  value can be passed to `ChromicPDF.print_to_pdf/2`.
+
+  ## Options
+
+  * `header`
+  * `footer`
+  * all options from `styles/1`
+
+  ## Example
+
+  This example has the dimension of a ISO A4 page.
+
+      ChromicPDF.Template.source_and_options(
+        html: "<p>Hello</p>",
+        header: "<p>header</p>",
+        footer: "<p>header</p>"
+        width: "210mm",
+        height: "297mm",
+        header_height: "45mm",
+        header_font_size: "20pt",
+        footer_height: "40mm"
+      )
+  """
+  @spec source_and_options([content_option() | style_option()]) ::
+          ChromicPDF.Processor.source_and_options()
+  def source_and_options(opts) do
+    content = Keyword.get(opts, :html, @default_content)
+    header = Keyword.get(opts, :header, "")
+    footer = Keyword.get(opts, :footer, "")
+    styles = styles(opts)
+
+    %{
+      source: {:html, [styles, content]},
+      opts: [
+        print_to_pdf: %{
+          preferCSSPageSize: true,
+          displayHeaderFooter: true,
+          headerTemplate: [styles, header],
+          footerTemplate: [styles, footer]
+        }
+      ]
+    }
+  end
+
+  @styles """
+  <style>
+    @page {
+      width: <%= @width %>;
+      height: <%= @height %>;
+      margin: <%= @header_height %> 0 <%= @footer_height %>;
+    }
+
+    #header {
+      padding: 0 !important;
+      height: <%= @header_height %>;
+      font-size: <%= @header_font_size %>;
+    }
+
+    #footer {
+      padding: 0 !important;
+      height: <%= @footer_height %>;
+      font-size: <%= @footer_font_size %>;
+    }
+
+    body {
+      margin: 0;
+      padding: 0;
+    }
+  </style>
+  """
+
+  @doc """
+  Renders page styles for given options.
+
+  These base styles will configure page dimensions and header and footer heights. They also
+  remove any browser padding and margins from these elements, and set the font-size.
+
+  If you want to use these, make sure to set the `preferCSSPageSize: true` option, or use
+  `source_and_options/1`.
+
+  ## Options
+
+  * `width` page width in any CSS unit, default: 279.4mm / 11 inches (US letter)
+  * `height` default: 215.9mm / 8.5 inches
+  * `header_height` default: zero
+  * `header_font_size` default: 10pt
+  * `footer_height` default: zero
+  * `footer_font_size` default: 10pt
+  """
+  @spec styles([style_option()]) :: blob()
+  def styles(opts \\ []) do
+    assigns = [
+      height: Keyword.get(opts, :height, "215.9mm"),
+      width: Keyword.get(opts, :width, "279.4mm"),
+      header_height: Keyword.get(opts, :header_height, "0"),
+      header_font_size: Keyword.get(opts, :header_font_size, "10pt"),
+      footer_height: Keyword.get(opts, :footer_height, "0"),
+      footer_font_size: Keyword.get(opts, :footer_font_size, "10pt")
+    ]
+
+    do_render_styles(assigns)
+  end
```

Comment:
Mmhh 🤔 
This seems pretty strict to me. Basically you can customize only height and font size of both header and footer, do I understand well? What if you have a more convoluted footer with different texts?
Do you need to style it inline?

Reply:
Yes, the idea here, or the current functionality of `Template`, is only to give you configuration + absolutely minimum styles to easily create a page that has visible & consistent header & footer, and no weird other paddings/margins anywhere. It's surprisingly difficult to do, at first :)

All client styles and more layout styles need to be passed to `:html`, `:header`, `:footer` as `<style>` tags. I'll work on the documentation a bit.

thanks for your review 💚